Source/Adobe.Target.Client/Util/CookieUtils.cs

```diff
@@ -0,0 +1,153 @@
+            var targetCookieValue = new StringBuilder();
+            long maxAge = 0;
+            maxAge = CreateSessionId(sessionId, nowInSeconds, targetCookieValue, maxAge);
+            maxAge = CreateDeviceId(deviceId, nowInSeconds, targetCookieValue, maxAge);
+            var cookieValue = targetCookieValue.ToString();
+
+            return string.IsNullOrEmpty(cookieValue) ? null : new TargetCookie(TargetConstants.MboxCookieName, cookieValue, (int)(maxAge / 1000));
+        }
+
+        internal static TargetCookie? CreateClusterCookie(string tntId)
+        {
+            if (tntId == null)
+            {
+                return null;
+            }
+
+            var locationHint = LocationHintFromTntId(tntId);
+
+            if (string.IsNullOrEmpty(locationHint))
+            {
+                return null;
+            }
+
+            long maxAge = GetNowInSeconds() + ClusterLocationHintMaxAge;
+
+            return new TargetCookie(TargetConstants.ClusterCookieName, locationHint, (int)(maxAge / 1000));
+        }
+
+        private static int GetNowInSeconds()
+        {
+            return (int)(DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() / 1000);
+        }
+
+        private static long CreateDeviceId(string deviceId, int nowInSeconds, StringBuilder targetCookieValue, long maxAge)
+        {
+            if (string.IsNullOrEmpty(deviceId))
+            {
+                return maxAge;
+            }
+
+            long deviceIdMaxAge = nowInSeconds + DeviceIdCookieMaxAge;
+            maxAge = Math.Max(maxAge, deviceIdMaxAge);
+            AppendCookieValue(deviceId, targetCookieValue, deviceIdMaxAge, TargetConstants.DeviceIdCookieName);
+
+            return maxAge;
+        }
+
+        private static long CreateSessionId(string sessionId, int nowInSeconds, StringBuilder targetCookieValue, long maxAge)
+        {
+            if (string.IsNullOrEmpty(sessionId))
+            {
+                return maxAge;
+            }
+
+            long sessionIdMaxAge = nowInSeconds + SessionIdCookieMaxAge;
+            maxAge = sessionIdMaxAge;
+            AppendCookieValue(sessionId, targetCookieValue, sessionIdMaxAge, TargetConstants.SessionIdCookieName);
+
+            return maxAge;
+        }
+
+        private static void AppendCookieValue(string id, StringBuilder targetCookieValue, long maxAge, string cookieName)
+        {
+            targetCookieValue.Append(cookieName)
+                .Append(InternalCookieSerializationSeparator)
+                .Append(id)
+                .Append(InternalCookieSerializationSeparator)
+                .Append(maxAge)
+                .Append(CookieValueSeparator);
+        }
+
+        private static TargetCookie? DeserializeInternalCookie(string cookie)
+        {
+            var cookieTokens = cookie.Split(InternalCookieSerializationSeparator);
+            if (cookieTokens.Length != 3)
+            {
+                return null;
+            }
+
```

Comment:
This might result in a format exceptions, I would suggest using `TryParse()` and us proper defensive coding.

Reply:
Will fix, thx!